### PR TITLE
fix: add a new setting to ignore certain folders while indexing

### PR DESCRIPTION
**Pull request by milesvdw**
Pretty much in the title. 
- add a new setting allowing users to configure a comma separated list of folder paths
- if a file's path starts with any of those folder paths, ignore it during indexing

This is important because sometimes I have folders with really eye-catching garbage that the AI _reallllly_ wants to use (like, templates that _look_ very similar to many other files, but don't actually have any content in them) and I don't want the Vault RAG to pick those up.

### CODE REVIEW

**Comment by logancyang** [comment]
This is an existing command

SCR-20240829-nppr
SCR-20240829-npmm

There's no centralized documentation now and that's on me.

If you'd like it to be a setting instead of a command, you might have to wait until I release mobile support and model revamp.

**Comment by milesvdw** [comment]
I'll likely just run what I've got locally for now, and when the next release comes I'll circle back. I didn't realize there was a command for it... I was poking around in the settings because that made the most sense to me... How could I see which folders I'd already excluded? I do definitely think it should be visible/persisted somewhere. The other LLM/vectorization/RAG plugins that I tried (I like yours the best) all had a similar setting.

File: src/constants.ts [hunk 1]
```diff
@@ -196,6 +196,7 @@ export const DEFAULT_SETTINGS: CopilotSettings = {
   lmStudioBaseUrl: "http://localhost:1234/v1",
   stream: true,
   defaultSaveFolder: "copilot-conversations",
+  excludedFolders: [],
   indexVaultToVectorStore: VAULT_VECTOR_STORE_STRATEGY.ON_MODE_SWITCH,
   qaExclusionPaths: "",
```
Comment (logancyang) [inline src/constants.ts]: It is right here. @milesvdw

File: src/constants.ts [hunk 2]
```diff
@@ -196,6 +196,7 @@ export const DEFAULT_SETTINGS: CopilotSettings = {
   lmStudioBaseUrl: "http://localhost:1234/v1",
   stream: true,
   defaultSaveFolder: "copilot-conversations",
+  excludedFolders: [],
```
Comment (logancyang) [inline src/constants.ts]: No need for this new setting.

**Comment by logancyang** [comment]
> I'll likely just run what I've got locally for now, and when the next release comes I'll circle back. I didn't realize there was a command for it... I was poking around in the settings because that made the most sense to me... How could I see which folders I'd already excluded? I do definitely think it should be visible/persisted somewhere. The other LLM/vectorization/RAG plugins that I tried (I like yours the best) all had a similar setting.

The setting exists and it's set via the command. You can check by opening the command modal, the field shows existing paths you set.

**Review by logancyang — CHANGES_REQUESTED** [review]
Can you use the existing setting field, and move the new setting to the QA setting block. Then it should be good to go.

**Comment by milesvdw** [comment]
> Can you use the existing setting field, and move the new setting to the QA setting block. Then it should be good to go.

@logancyang thanks! I made the updates/simplifications as requested. Hopefully this setting being visible is helpful for other people! Loving the plugin, I use it every day.

**Comment by logancyang** [comment]
Can you fix the build error. I recommend testing with `npm run build`, as well as `npm run dev` and manual checks in the app before submitting a PR.

**Comment by milesvdw** [comment]
> Can you fix the build error. I recommend testing with `npm run build`, as well as `npm run dev` and manual checks in the app before submitting a PR.

@logancyang ah yes, apologies for the lax QA. I didn't realize you were storing your folders as a string instead of the list of strings I was using. All fixed now, and it does appear to be working with the current build in my local repo.

**Comment by milesvdw** [comment]
> Owner
> ###

@logancyang ah yes, apologies for the lax QA. I didn't realize you were storing your folders as a string instead of the list of strings I was using. All fixed now, and it does appear to be working with the current build in my local repo.

**Comment by milesvdw** [comment]
> Can you fix the build error. I recommend testing with `npm run build`, as well as `npm run dev` and manual checks in the app before submitting a PR.

@logancyang ah yes, apologies for the lax QA. I didn't realize you were storing your folders as a string instead of the list of strings I was using. All fixed now, and it does appear to be working with the current build in my local repo.

**Comment by milesvdw** [comment]
> Can you fix the build error. I recommend testing with `npm run build`, as well as `npm run dev` and manual checks in the app before submitting a PR.

@logancyang ah yes, apologies for the lax QA. I didn't realize you were storing your folders as a string instead of the list of strings I was using. All fixed now, and it does appear to be working with the current build in my local repo.

File: src/main.ts [hunk 3]
```diff
@@ -545,6 +545,7 @@ export default class CopilotPlugin extends Plugin {
           content: fileContents[index],
           metadata: fileMetadatas[index]?.frontmatter ?? {},
         };
+
```
Comment (logancyang) [inline src/main.ts]: No need to change this file?

File: src/settings/components/GeneralSettings.tsx [hunk 4]
```diff
@@ -79,6 +79,13 @@ const GeneralSettings: React.FC<GeneralSettingsProps> = ({
         value={settings.defaultSaveFolder}
         onChange={(value) => updateSettings({ defaultSaveFolder: value })}
       />
+      <TextComponent
```
Comment (logancyang) [inline src/settings/components/GeneralSettings.tsx]: This should be in QA settings instead.

Comment (milesvdw) [reply src/settings/components/GeneralSettings.tsx]: sure - I assume you'll move this around whenever the larger feature is out of QA?
also fixed a small typo I spotted in qa settings.

@logancyang anything else you need? I assume you'll just merge this whenever you feel like, I plan to forget about it (but will definitely use the UX feature when released with the plugin!)

**Review by logancyang — APPROVED** [review]

**Comment by logancyang** [comment]
LGTM, thanks for the contribution!

**Pull request merged**